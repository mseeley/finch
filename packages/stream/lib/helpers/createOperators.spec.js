/* eslint-env jest */

const { localNameOf } = require("@finch/core");
const createOperators = require("./createOperators");

describe(localNameOf(__filename), () => {
  const use = "./__fixtures__/operators/atypical/empty";

  it("returns nested stages as flat array of operators", () => {
    // The actual stagrs are unimportant, focus on the return structure.
    const stages = [{ use }, [{ use }, [{ use }]]];
    const operators = createOperators(stages);

    expect(operators).toMatchSnapshot();
  });
});
