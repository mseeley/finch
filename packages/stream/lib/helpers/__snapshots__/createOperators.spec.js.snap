// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stream/lib/helpers/createOperators.spec returns nested stages as flat array of operators 1`] = `
Array [
  [Function],
  [Function],
  [Function],
]
`;
